packages/wrangler/templates/startDevWorker/InspectorProxyWorker.ts

```diff
@@ -0,0 +1,472 @@
+	WRANGLER_VERSION: string;
+	DURABLE_OBJECT: DurableObjectNamespace;
+}
+
+export default {
+	fetch(req, env) {
+		const singleton = env.DURABLE_OBJECT.idFromName("");
+		const inspectorProxy = env.DURABLE_OBJECT.get(singleton);
+
+		return inspectorProxy.fetch(req);
+	},
+} as ExportedHandler<Env>;
+
+export class InspectorProxyWorker implements DurableObject {
+	constructor(_state: DurableObjectState, readonly env: Env) {}
+
+	#websockets: {
+		proxyController?: WebSocket;
+		runtime?: WebSocket;
+		devtools?: WebSocket;
+	} = {};
+	#runtimeWebSocketPromise = createDeferredPromise<WebSocket>();
+
+	#proxyData?: ProxyData;
+
+	runtimeMessageBuffer: string[] = [];
+
+	#handleRuntimeIncomingMessage = (event: MessageEvent) => {
+		assert(typeof event.data === "string");
+		assert(
+			this.#websockets.proxyController !== undefined,
+			"Expected proxy controller websocket"
+		);
+
+		const msg = JSON.parse(event.data) as
+			| DevToolsCommandResponses
+			| DevToolsEvents;
+		this.sendDebugLog("RUNTIME INCOMING MESSAGE", msg);
+
+		if (
+			"method" in msg &&
+			(msg.method === "Runtime.exceptionThrown" ||
+				msg.method === "Runtime.consoleAPICalled")
+		) {
+			this.sendProxyControllerMessage(event.data);
+		}
+
+		this.runtimeMessageBuffer.push(JSON.stringify(msg));
+
+		this.#tryDrainRuntimeMessageBuffer();
+	};
+
+	#tryDrainRuntimeMessageBuffer = () => {
+		if (this.#websockets.devtools === undefined) return;
+
+		// clear the buffer and replay each message to devtools
+		for (const data of this.runtimeMessageBuffer.splice(0)) {
+			this.sendDevToolsMessage(data);
+		}
+	};
+
+	#runtimeMessageCounter = 1e8;
+	nextCounter() {
+		return ++this.#runtimeMessageCounter;
+	}
+	keepAliveInterval: number | null = null;
+	#handleProxyControllerIncomingMessage = (event: MessageEvent) => {
+		assert(
+			typeof event.data === "string",
+			"Expected event.data from proxy controller to be string"
+		);
+
+		const message: InspectorProxyWorkerIncomingWebSocketMessage = JSON.parse(
+			event.data
+		);
+
+		this.sendDebugLog("handleProxyControllerIncomingMessage", event.data);
+
+		switch (message.type) {
+			case "reloadComplete":
+				this.#proxyData = message.proxyData;
+
+				this.reconnectRuntimeWebSocket();
+				this.notifyDevToolsOfReloadComplete();
+
+				break;
+			default:
+				assertNever(message.type);
+		}
+	};
+	notifyDevToolsOfReloadComplete() {
+		// // If devtools already has sources loaded, they may now be stale
+		// // TODO: figure out how to get devtools to reload
+		// this.#websockets.devtools?.close();
+		//
+		// this.sendDevToolsMessage({
+		// 	method: "Runtime.executionContextDestroyed",
+		// 	params: {
+		// 		executionContextId: 1,
+		// 		executionContextUniqueId:
+		// 			this.latestExecutionContextCreatedMessage.context
+		// 				.executionContextUniqueId,
+		// 	},
+		// });
+		//
+		// this.sendRuntimeMessage({
+		// 	method: "Runtime.enable",
+		// 	id: this.nextCounter(),
+		// });
+	}
```

Comment:
This is a remnant of something were trying to fix a bug we discovered via the devtools: the Source panel files do not update across reloads (source file changes) – the devtools doesn't know the runtime reloaded because the executionContextId always starts at `1` across reloads

This has been fixed downstream in workerd by setting the executionContextId to a random number  in https://github.com/cloudflare/workerd/pull/1178